errgroup.go

```diff
@@ -204,16 +221,30 @@ func (g *Group) startG() {
 				return
 			}
 
-			if err := f(); err != nil {
```

Comment:
This looks nicer but as explained above we are already needing the `RWMutex` so this can be achieved by using the RWMutex instead of also introducing a new field `sync.Once`. 

So let me know if you think its better to stay using sync.Once in addition to the RWMutex just for readability.